relayer/src/worker/client.rs

```diff
@@ -130,7 +130,11 @@ impl ClientWorker {
         }
     }
 
-    fn detect_misbehaviour(&self, client: &ForeignClient, update: Option<UpdateClient>) -> bool {
+    fn detect_misbehaviour(
```

Comment:
Functions like this is why we need generic types to find out when the position of the identifiers get swapped. Notice that we are in a `ClientWorker<ChainA, ChainB>`, but the `process_cmd` function requires a `ForeignClient<ChainB, ChainA>`, making `ChainB` the host chain here.